### PR TITLE
Fix RepairJobsTest.testNoTreesRetainedAfterDifference by waiting for …

### PR DESCRIPTION
**Pull request by jasonstack**
…latch

### CODE REVIEW

File: test/unit/org/apache/cassandra/repair/RepairJobTest.java [hunk 1]
```diff
@@ -152,6 +182,7 @@ public void setup()
     @After
     public void reset()
     {
+        session.onSyncComplete.clear();
```
Comment (maedhroz) [inline test/unit/org/apache/cassandra/repair/RepairJobTest.java]: nit: Not entirely necessary, given we create a new session every test w/ `@Before`?

File: test/unit/org/apache/cassandra/repair/RepairJobTest.java [hunk 2]
```diff
@@ -91,14 +95,16 @@
...
-    private RepairSession session;
+    private MeasureableRepairSession session;
     private RepairJob job;
     private RepairJobDesc sessionJobDesc;
 
     // So that threads actually get recycled and we can have accurate memory accounting while testing
     // memory retention from CASSANDRA-14096
     private static class MeasureableRepairSession extends RepairSession
     {
+        List<Callable<?>> onSyncComplete = new ArrayList<>();
```
Comment (maedhroz) [inline test/unit/org/apache/cassandra/repair/RepairJobTest.java]: nit: Make it `final`
naming nit: perhaps something like `syncCompleteCallbacks` or `completionCallbacks`

**Review by maedhroz — APPROVED** [review]
Just left a couple nits, but otherwise, LGTM

File: test/unit/org/apache/cassandra/repair/RepairJobTest.java (hunk 2 above)
Comment (jasonstack) [reply test/unit/org/apache/cassandra/repair/RepairJobTest.java]: +1 changed to `syncCompleteCallbacks`

File: test/unit/org/apache/cassandra/repair/RepairJobTest.java (hunk 1 above)
Comment (jasonstack) [reply test/unit/org/apache/cassandra/repair/RepairJobTest.java]: removed

**Comment by maedhroz** [comment]
+1

**Pull request closed without merging**